File: LeetCode/Python/Easy/1512_number_of_good_pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        d = {}
        ans = 0
        for num in nums:
            if num not in d:
                d[num] = 0
            d[num] += 1
        for key, val in d.items():
            ans += (val * (val - 1)) // 2
        return ans

File: LeetCode/Javascript/Easy/1512_number_of_good_pairs.js
const numIdenticalPairs = nums => {
    const counts = {};
    let ans = 0;
    for(let i=0; i < nums.length; i++){
        if(nums[i] in counts)
            counts[nums[i]]++;
        else
            counts[nums[i]] = 1;
    }
    for (let [key, value] of Object.entries(counts)) {
        ans += value * (value - 1) / 2;
    }
    return ans;
};